### ISSUE TITLE
Not receiving data

### ISSUE THREAD

**Issue by DanielXiaomi** [flavio20002/mitemp_bt#1]
Hi,

so i installed hass.io on a pi3 and added _mitemp_bt_ to my sensors in yaml. Unfortunatelly I only get  an "Updating mitemp_bt sensor took longer than the scheduled" for both of my xiaomi sensor. 

This is my config: 
`sensor:
  - platform: yr
  - platform: mitemp_bt
    mac: '4c-65-a8-d4-c5-xx'
    name: Living Room Temp
    force_update: false
    median: 3
    monitored_conditions:
      - temperature
      - humidity

  - platform: mitemp_bt
    mac: '4c-65-a8-d5-41-xx'
    name: Hall Temp
    force_update: true
    median: 1
    monitored_conditions:
      - temperature
      - humidity`

As you can see, neither false nor true force_update is working. What am I doing wrong?

**Comment by flavio20002** [comment]
Try to use the official one, instead of my sensor.

https://www.home-assistant.io/components/sensor.mitemp_bt/

If also the official one doesn't work, then you should troubleshoot your bluetooth connection. Maybe you have a bluetooth device tracker that could interfere, or maybe the sensors are too far away?

**Comment by DanielXiaomi** [comment]
Thanks for your fast answer. 
Sorry, I thought you were the original author of this component. 
So obviously I use the official one, since I didnt install anythiny else. Nevertheless its not working. A Bluetooth tracker is not in use. My config is based on https://www.home-assistant.io/components/sensor.mitemp_bt/

**Comment by flavio20002** [comment]
I don't know Hass.io, but do you have file system access to Home assistant directory? If so, search for the file mitem_bt.py, rename in .old and move there my file with the same name. Restart the Home ssistant software and then you could have a chance to use my component.

**Comment by DanielXiaomi** [comment]
Indeed I have file system access, but hass.io has no intern search, so i have to find the py-file manually. Do you have a hint where it might be located?

**Comment by flavio20002** [comment]
In a standard installation, the file is located in /srv/homeassistant/lib/python3.5/site-packages/homeassistant/components/sensor but I don't know how Hass.io structure files...I'm sorry

**Comment by DanielXiaomi** [comment]
Well, hass.io is definatelly different from a standard installation, so i did the following way: 
https://developers.home-assistant.io/docs/en/creating_component_loading.html.

I copied your component and named it "mitemp_bt_new.py" and saved it to "<config directory>/custom_components/sensormitemp_bt_new.py". 
In the yaml i typed: 
sensor:
  - platform: yr
  - platform: mitemp_bt_new
    mac: '4c-65-a8-d4-c5-xx'
    name: Livingroom Temp
    force_update: false
    median: 3
    monitored_conditions:
      - temperature
      - humidity

So after a restart I still dont receive data, but i can see "Updating mitemp_bt_new sensor took longer than the scheduled update interval 0:00:30" in the log file. 

I think i'll just bury my project here. This took too much of my sunday morning. 
But thank you flavio for your kind help. 


Edit: There must be an error with bluetooth, because the config definatelly uses your component since I can see "Umidità" in my Dashboard and not "humidity" anymore.

**Comment by DanielXiaomi** [comment]
I finaly got the solution, after a sleep and 2 coffees. Maybe I can help someone else with this looking for a solution opon the internet. 

1. I made a big mistake. I copied blind the mac from my BLE sniffer (xx-xx-xx). After reading the mitemp_bt.py I saw it must be xx:xx:xx, so I changed that. -Restart- _Still didnt work out._ 

2. After some google "research" I found out one should delete the device from within the xiaomi mi app. I restarted hass.io. From now on, hass.io found the devices, but still didnt show any data. 

3. I renamed the names from "Livingroom Temp" to "LivingroomTemp" (no more space left). And after a reboot it worked like a charme. All data is now shown on my dashboard.
